kitura-init.js

```diff
@@ -1,32 +1,148 @@
 #!/usr/bin/env node
 
 const chalk = require('chalk');
-const spawn = require('child_process').spawn;
+const spawnSync = require('child_process').spawnSync;
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
 
-let args = process.argv.slice(2);
+const initURL = 'https://github.com/IBM-Swift/generator-swiftserver-projects'
+const initBranch = 'init'
 
-let options = ['-p', 'yo@1', '-p', 'generator-swiftserver', '--', 'yo', 'swiftserver'];
+let args = process.argv.slice(2);
 
 if (args.length > 0) {
     if (args.indexOf('--help') > -1) {
         printHelp();
         process.exit(0);
     }
+}
+
+let currentDirPath = path.resolve("./");
+let currentDir = path.basename(currentDirPath);
+// Replace spaces with hyphens so Xcode project will build.
+let projName = currentDir.replace(/ /g,"-");
 
-    // Add on the arguments
-    options.push(args);
+if (projName.charAt(0) === '.') {
+    console.error(chalk.red('Application name cannot start with .: %s'));
+   }
+if (projName.toLowerCase() === 'node_modules') {
+    console.error(chalk.red('Application name cannot be {node_modules}'));
+   }
+if (projName.toLowerCase() === 'favicon.ico') {
```

Comment:
This is a bit of an odd check - can you explain to me how it came about?

Reply:
It was something that came from the generator so assumed it must be there for a reason... Although I'm sure there are more names which would break things, so maybe we just take it out and just leave in the first case (not allowing anything which starts with a ".")